File: stalin_sort.scala
import scala.collection.mutable.ListBuffer
import scala.util.Random

def stalin_sort(arr: ListBuffer[Double]): Array[Double] =
    if arr.length == 0 then
        return arr.toArray
    
    var max_v = arr.head
    var sorted = new ListBuffer[Double]
    for i <- arr do
        if i >= max_v then
            sorted += i
            max_v = i
        else
            println(s"$i sent to Gulag")
    var sorted_li = sorted.toArray
    return sorted_li

end stalin_sort

@main def main(): Unit = {
    var ss = new ListBuffer[Double]
    var t = 0
    while t < 50 do
        var rr = 100 * Random.nextDouble()
        ss += rr
        t = t + 1
    var srr = ss.toArray
    println("[" + srr.mkString(", ") + "]")
    println("[" + stalin_sort(ss).mkString(", ") + "]")
}

File: stalin_sort.rb
def stalin_sort(arr)
  if arr.size == 0
    return arr
  end
  max_v = arr[0]
  sorted = []
  arr.each do |i|
    if i >= max_v
      sorted.push(i)
      max_v = i
    else
      puts "#{i} sent to Gulag"
    end
  end
  return sorted
end

arrc = []
i = 0
while i < 50
  iw = Random.rand(100.0)
  arrc.push(iw)
  i += 1
end

puts arrc.inspect
puts stalin_sort(arrc).inspect
